clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -221,16 +223,16 @@ public RecordAppendResult append(TopicPartition tp,
                 if (closed)
                     throw new KafkaException("Producer closed while send in progress");
 
-                RecordAppendResult appendResult = tryAppend(timestamp, key, value, headers, callback, dq);
+                RecordAppendResult appendResult = tryAppend(timestamp, key, value, headers, callback, dq, nowMs);
```

Comment:
I don't think it's safe to reuse `nowMs` here since we may have blocked on `free.allocate`.

Reply:
Done.